library/mt_login_test.py

```diff
@@ -23,12 +22,19 @@ def main():
   msg = ""
 
   mk = mt_api.Mikrotik(hostname,username,password)
-  try:
-    mk.login()
-  except:
-    module.fail_json(
-      msg="Could not log into Mikrotik device.  Check the username and password."
-    )
+  sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+  result = sock.connect_ex((hostname, 8728))
+  if result == 0:
+    try:
+      mk.login()
+    except:
+      module.fail_json(
+        msg="Could not log into Mikrotik device.  Check the username and password."
+      )
+  else:
+      module.fail_json(
+        msg="Could not access RouterOS api." + " Verify API service is enabled and not blocked by firewall."
```

Comment:
no idea why this only works against real hardware... wonder what the difference is between the vm and the real hardware.